statistics-dashboard-service/src/main/java/eu/europeana/statistics/dashboard/service/StatisticsService.java

```diff
@@ -150,7 +166,17 @@ private StatisticsQuery prepareFilteringQuery(StatisticsFilteringRequest statist
 
   private Map<MongoStatisticsField, ValueRange> prepareRangeFilteringOptionsQuery(StatisticsQuery query) {
     Map<MongoStatisticsField, ValueRange> result = new EnumMap<>(MongoStatisticsField.class);
-    MongoStatisticsField.getRangeFields().forEach(field -> result.put(field, query.queryForValueRange(field)));
+    MongoStatisticsField.getRangeFields().forEach(field ->
+        {
+            ValueRange valueRange;
+            if (query.queryForValueRange(field) == null){
```

Comment:
Why is null not okay?
And if it really isn't okay then perhaps the `new ValueRange("", ""); ` would better happen in the function itself.

Reply:
After some clarification, we decided on changing the original method to return `Optional<ValueRange>`. I will add those changes